core/Network.js

```diff
@@ -86,12 +87,20 @@ Network.prototype.getProcessByName = function (processName) {
 
 Network.prototype.run = function (runtime, options, callback) {
   options = options || {};
-
-  _.forEach(this._processes, function (process) {
-    _.invokeMap(process.inports, 'setRuntime', runtime);
-    _.invokeMap(process.outports, 'setRuntime', runtime);
-  });
-  runtime.run(_.values(this._processes), options, callback || function () {});
+  callback = callback || function () {};
+
+  _.invokeMap(this._connections, 'setRuntime', runtime);
+
+  try {
+    runtime.run(_.values(this._processes), options, callback );
+  } catch (e) {
+    console.log('Connections');
```

Comment:
Sounds good - is this integrated with my deadlock display - or does it
replace it?!

On Wed, Jun 8, 2016 at 11:04 AM, Dan Rumney <notifications@github.com>
wrote:

> In core/Network.js
> <https://github.com/jpaulm/jsfbp/pull/74#discussion_r66273523>:
>
> > @@ -86,12 +87,20 @@ Network.prototype.getProcessByName = function (processName) {
> >
> >  Network.prototype.run = function (runtime, options, callback) {
> >    options = options || {};
> > -
> > -  _.forEach(this._processes, function (process) {
> > -    _.invokeMap(process.inports, 'setRuntime', runtime);
> > -    _.invokeMap(process.outports, 'setRuntime', runtime);
> > -  });
> > -  runtime.run(_.values(this._processes), options, callback || function () {});
> > +  callback = callback || function () {};
> > +
> > +  _.invokeMap(this._connections, 'setRuntime', runtime);
> > +
> > +  try {
> > +    runtime.run(_.values(this._processes), options, callback );
> > +  } catch (e) {
> > +    console.log('Connections');
>
> I am finding this very useful. When there's a deadlock, show all the
> connections and which ones are full. This may help identify *why* there's
> a deadlock.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jpaulm/jsfbp/pull/74/files/d12c2b94b326314da1d5fe633ff423f329d25d0b#r66273523>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AATGJzMTFg7DyGNxVKEd04O0zZJkEXXbks5qJtoOgaJpZM4IviIh>
> .
>